src/traffic_server/traffic_server.cc

```diff
@@ -779,14 +788,13 @@ CB_After_Cache_Init()
   }
 #endif
 
-  if (1 == start) {
-    // The delay_listen_for_cache value was 1, therefore the main function
-    // delayed the call to start_HttpProxyServer until we got here. We must
-    // call accept on the ports now that the cache is initialized.
+  if (CacheWaitState::DELAYED == cache_wait_state) {
+    // Initialization was delayed to wait for cache. Cache is now initialized and therefore
+    // it's time to get the party started.
     Debug("http_listen", "Delayed listen enable, cache initialization finished");
     start_HttpProxyServer();
-    emit_fully_initialized_message();
   }
+  emit_fully_initialized_message();
```

Comment:
Hmm why is this called multiple times? It's called again at line 2183.

Reply:
Because only one of these should actually happen.